tests/test_smoke.py

```diff
@@ -2297,6 +2323,29 @@ def test_aws_zero_quota_failover():
     run_one_test(test)
 
 
+@pytest.mark.gcp
+def test_gcp_zero_quota_failover():
+
+    name = _get_cluster_name()
+    region = get_gcp_region_for_quota_failover()
+
+    if not region:
+        warnings.warn(
+            f'Unable to test zero quota failover optimization — quotas '
+            f'for A100 GPUs were found on all GCP regions. Is this '
+            f'expected for your account?')
+        return
```

Comment:
I found this warning is never printed and my tests were silently passing. In general, it seems like `warnings.warn` is not printed when running pytest with xdist.

Can we instead xfail this test with the same message?

```suggestion
        pytest.xfail(
            'Unable to test zero quota failover optimization — quotas '
            'for A100 GPUs were found on all GCP regions. Is this '
            'expected for your account?')
```

Reply:
And perhaps do the same for AWS